scr/main/java/bittercode/model/User.java

```diff
@@ -69,20 +69,22 @@ public void setRoles(List<UserRole> roles) {
         this.roles = roles;
     }
 
-//    public static User retrieveFromHttpServletRequest(HttpServletRequest req) {
-//        User user = new User();
-//        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
-//        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
-//        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
-//        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
-//        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
-//        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
-//        user.setEmailId(mailId);
-//        user.setFirstName(fName);
-//        user.setLastName(lName);
-//        user.setPassword(pWord);
-//        user.setPhone(Long.parseLong(phNo));
-//        user.setAddress(addr);
-//        return user;
-//    }
-}
+    public static User retrieveFromHttpServletRequest(HttpServletRequest req) {
+        User user = new User();
+        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
+        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
+        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
+        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
+        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
+        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
+
+        user.setEmailId(mailId);
+        user.setFirstName(fName);
+        user.setLastName(lName);
+        user.setPassword(pWord);
+        user.setPhone(Long.parseLong(phNo));
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Add error handling for number parsing</b></div>
<div id="fix">

Consider adding try-catch block around `Long.parseLong()` to handle potential `NumberFormatException`
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -85 +85,5 @@
 -        user.setPhone(Long.parseLong(phNo));
 +        try {
 +            user.setPhone(Long.parseLong(phNo));
 +        } catch (NumberFormatException e) {
 +            throw new IllegalArgumentException("Invalid phone number format", e);
 +        }
```


```suggestion
        try {
            user.setPhone(Long.parseLong(phNo));
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Invalid phone number format", e);
        }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/6#issuecomment-2527697240>#0a012d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged